File: nginx/src/script/object/player.js
import * as THREE from 'three';
import { websocket } from "../network.js";

export class Player {
	constructor(position, color, name, up, down) {
		const geometry = new THREE.BoxGeometry(0.1, 0.8, 0.1);
		const material = new THREE.MeshBasicMaterial({ color: color });
		this.mesh = new THREE.Mesh(geometry, material);
		this.mesh.position.set(position, 0, 0);

		this.name = name;
		this.keyMap = { up: up, down: down };
		this.keyInput = { up: false, down: false };
		this.score = 0;

		this.velocity = 0;
		this.maxVelocity = 15;
		this.acceleration = 10;
		this.deceleration = 10;
		this.currentAcceleration = 0;
		this.maxAcceleration = this.acceleration;
		this.accelerationRate = 2;
	}

	keyDown = (event) => {
		if (event.key === this.keyMap.up) {
			this.keyInput.up = true;
		} else if (event.key === this.keyMap.down) {
			this.keyInput.down = true;
		}
	}
	keyUp = (event) => {
		if (event.key === this.keyMap.up) {
			this.keyInput.up = false;
		} else if (event.key === this.keyMap.down) {
			this.keyInput.down = false;
		}
	}
	input() {
		window.addEventListener("keydown", this.keyDown);
		window.addEventListener("keyup", this.keyUp);
	}
	move(deltatime) {
		const limit = 3.3;
		if (this.keyInput.up && this.mesh.position.y < limit) {
			this.currentAcceleration = Math.min(this.currentAcceleration + this.accelerationRate * deltatime, this.maxAcceleration);
			this.velocity = Math.min(this.velocity + this.acceleration, this.maxVelocity);
		} else if (this.keyInput.down && this.mesh.position.y > -limit) {
			this.currentAcceleration = Math.min(this.currentAcceleration + this.accelerationRate * deltatime, this.maxAcceleration);
			this.velocity = Math.max(this.velocity - this.acceleration, -this.maxVelocity);
		} else if (Math.abs(this.velocity) > 0) {
			this.currentAcceleration = 0;
			this.velocity *= 0.9;
			if (Math.abs(this.velocity) < 0.001) {
				this.velocity = 0;
			}
		}
		this.mesh.position.y += this.velocity * deltatime;
		if (this.mesh.position.y > limit) {
			this.mesh.position.y = limit;
			this.velocity = 0;
		} else if (this.mesh.position.y < -limit) {
			this.mesh.position.y = -limit;
			this.velocity = 0;
		}
	}
	getBound() {
		const halfWidth = 0.05;
		const halfHeight = 0.4;
		return {
			minX: this.mesh.position.x - halfWidth,
			maxX: this.mesh.position.x + halfWidth,
			minY: this.mesh.position.y - halfHeight,
			maxY: this.mesh.position.y + halfHeight
		};
	}
	removeEvent() {
		window.removeEventListener("keydown", this.keyDown);
		window.removeEventListener("keyup", this.keyUp);
	}
}

File: django/src/game/player.py
from django.conf import settings

class Player:
    def __init__(self, position, name):
        self.name = name
        self._paddle = settings.PADDLE,
        self.position = { "x": position, "y": 0 }
        self._keyInput = { "up": False, "down": False }
        self.score = 0
        self._velocity = 0
        self._maxVelocity = 15
        self._acceleration = 10
        self._deceleration = 10
        self._currentAcceleration = 0
        self._maxAcceleration = self._acceleration
        self._accelerationRate = 2

    def input(self, input, value):
        self._keyInput[input] = value

    def move(self, deltatime):
        limit = settings.HEIGHT_LIMIT
        if self._keyInput["up"] and self.position["y"] < limit:
            self._currentAcceleration = min(self._currentAcceleration + self._accelerationRate * deltatime, self._maxAcceleration)
            self._velocity = min(self._velocity + self._acceleration, self._maxVelocity)
        elif self._keyInput["down"] and self.position["y"] > -limit:
            self._currentAcceleration = min(self._currentAcceleration + self._accelerationRate * deltatime, self._maxAcceleration)
            self._velocity = max(self._velocity - self._acceleration, -self._maxVelocity)
        elif abs(self._velocity) > 0:
            self._currentAcceleration = 0
            self._velocity *= 0.9
            if abs(self._velocity) < 0.001:
                self._velocity = 0
        
        self.position["y"] += self._velocity * deltatime
        if self.position["y"] > limit:
            self.position["y"] = limit
            self._velocity = 0
        elif self.position["y"] < -limit:
            self.position["y"] = -limit
            self._velocity = 0

    def getBound(self):
        halfWidth = 0.05
        halfHeight = 0.4
        return {
            "minX": self.position["x"] - halfWidth,
            "maxX": self.position["x"] + halfWidth,
            "minY": self.position["y"] - halfHeight,
            "maxY": self.position["y"] + halfHeight
        }
    
    def getJson(self):
        return {
            "position": self.position,
            "score": self.score
        }